lib/charms/vault_k8s/v0/vault_client.py

```diff
@@ -46,6 +75,19 @@ class Vault:
     def __init__(self, url: str, ca_cert_path: str):
         self._client = hvac.Client(url=url, verify=ca_cert_path)
 
+    def authenticate(self, auth_details: Union[Token, AppRole]) -> None:
```

Comment:
I've managed to avoid using protocols up until this point because I've found them esoteric, but I'm not opposed to that if we plan to support more auth methods in the long run.

Reply:
I did end up adding the protocol, not sure if you've seen it.